### ISSUE TITLE
Comment after if moved to if line lacks space before it

### ISSUE THREAD

**Issue by yebblies** [Hackerpilot/dfmt#71]
Also occurs with `else, `while`, `for` etc
```diff
unittest
{
-    if (0)
-        // comment
+    if (0)// comment
    {
    }
}
```

**Comment by Hackerpilot** [comment]
That is some weird code.

**Comment by yebblies** [comment]
Yeah, it's from a bug in magicport's pretty-printer, it should be putting the comment on the same line as the `if`.  A similar style does occur in real code like this though:
```d
if (x)
    // comment
    doSomething();
```
I really wish it didn't though, it make it hard to read and is related to the infamous
```d
if (x)
    //doSomething();
```

**Comment by Hackerpilot** [comment]
I think I'll make it output this:
```d
if (x) // comment
    doSomething();
```
and then call it fixed.

**Comment by yebblies** [comment]
That would be perfect.  What will happen if it's a multi-line comment?
eg
```d
if (x)
    /* blah blah
     * blah
     */
    doSomething();
```
I've been adding braces to these when I see them in the dmd source.

**Comment by yebblies** [comment]
A semi-related case is this
```d
int var1; /* this comment
           * applies to var1
           */
int var2; // but not this one
```
vs
```d
int var1; // this comment
          // applies to var1
int var2; // but not this one
```
Which magicport currently horrifically mangles, and is difficult to detect.

**Comment by Hackerpilot** [comment]
I think that fixing this case:
```d
int var1; // this comment
          // applies to var1
int var2; // but not this one
```
will be enough trouble that it's not worth fixing.

The original issue with the space being missing before the comment was fixed as part of #27, so I'm going to close this.

**Comment by yebblies** [comment]
> will be enough trouble that it's not worth fixing.

I think you're probably right.  I've been removing them from dmd's source when I find them.
